### ISSUE TITLE
Invalid case for conditional_binding_cascade

### ISSUE THREAD

**Issue by zetasq** [realm/SwiftLint#667]
My code is like below:

`if case .Success(let info) = response.result, let jsonDic = info as? [String: AnyObject]`

I can't remove the second "let" to fix this warning. The implementation of conditional_binding_cascade should be improved to remove this bug.

**Comment by lessthanyouthink** [comment]
I found a related invalid case when using enums with multiple associated values. For example:

`if case .Response(response: let r, json: let j) = result`

**Comment by jpsim** [comment]
that second one will be really tricky to fix @lessthanyouthink

**Comment by zwaldowski** [comment]
A few other false positives, if it helps:

```swift
guard case let .NetworkError(anotherError) = error, let nsUrlError = anotherError as? NSURLError else {
```

```swift
if case let .RunningPath(parentPath) = state, let restorationIdentifier = vc.restorationIdentifier
```

```swift
if case (let oldType, var indexPaths)? = inFlightItemTransactions where oldType == type
```

```swift
guard let sself = self, case .Success(let data) = imageResult else { return }
```

**Comment by paddlefish** [comment]
Fixed in #673 .  Actually it just bails when case is involved since it appears to be pretty hard to determine when let is in a case vs when it's in the expression.  (At least I couldn't find a way to glean that from the current info that the code analysis dictionary had.)
